qa/rpc-tests/test_framework/blocktools.py

```diff
@@ -39,15 +41,19 @@ def add_witness_commitment(block, nonce=0):
 
     # witness commitment is the last OP_RETURN output in coinbase
     output_data = WITNESS_COMMITMENT_HEADER + ser_uint256(witness_commitment)
-    block.vtx[0].vout.append(CTxOut(0, CScript([OP_RETURN, output_data])))
+    block.vtx[0].vout.append(CTxOut(CTxOutValue(0), CScript([OP_RETURN, output_data])))
     block.vtx[0].rehash()
     block.hashMerkleRoot = block.calc_merkle_root()
     block.rehash()
 
 
 def serialize_script_num(value):
-    r = bytearray(0)
     if value == 0:
+        return b'\x00'
+    # short form OP_1 - OP_16
+    r = bytearray()
+    if value > 0 and value < 17:
+        r.append(0x50 + value)
```

Comment:
previously this function would not produce the canonical encoding for values [1, 16]

Reply:
Not a script expert, but these values correspond to opcodes, not the resulting value pushed on the stack.

The code as-is matches the C++ code quite well.

Reply:
this function is used in exactly one place (line 69 to the left). Its result is passed to ``ser_string()`` and passed in as a script. Perhaps I'm missing something, but it's not clear to me how that could possibly generate a valid script (apart from the special case of 0 and possibly some others). [This](https://github.com/ElementsProject/elements/blob/elements-0.14.1/src/validation.cpp#L3872) check was failing in the test.

Reply:
that said, it's probably better to leave this function as-is, and instead wrap it in ``CScript()`` where it's used instead